### DISCUSSION TITLE
Time series regression: Forecasting models or Classification models ?

### DISCUSSION THREAD

**Discussion by Sinnaeve** [alan-turing-institute/sktime#3389] [Q&A]
Hi, I'm working on Time series regression.

For context: I'm trying to propose a set of solutions for time series regression in industrial environment for things like predicting failures of assets, predicting quality at the end of an industrial process...
The goal is to propose a starting point for teams that would work on this kind of projects later.

The issue I have is that I find that the literature focus more on either time series classification (with hive-cote, rocket...) or time series forecasting (with things like Arima, or more complex models like the ones here: https://pytorch-forecasting.readthedocs.io/en/stable/models.html).

Time series regression seems similar to both those tasks, but the models seems very different.
So I was wondering how to choose which of this approach to use if I want to do time series regression ? I imagine that is dependent of the specific problem, but is there things to look at to make the choice ? 
(like I would imagine that if I have time series with a clear period and static like Electrocardiography or motion capture data, may be time series classification inspired model would be more appropriate; while if I have data with trend and periodicity forecasting model are more appropriate)

**Comment by Sinnaeve** [comment]
What I found looking into it is that forecasting seems to give more importance to recent values compare to classification when making a prediction. So a regression problem (time series input to predict continuous output) would more close to time series classification when old data are as much relevant than very recent data, while problem when very recent data are more relevant would be more close to forecasting.

Does someone know if this is moslty correct ?

Source:
"a regression task of which the aim is to learn the relationship between a time series and a continuous scalar variable; a task closely related to time series classification. This task generalizes time series forecasting, relaxing the requirement that the value predicted be a future value of the input series or primarily depend on more recent values" from [Time series extrinsic regression
](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7951134/)

 "This (regression) differs slightly from time series forecasting (TSF) in which the goal is to predict future values in the sequence.[...] TSF usually assumes that the most recent values in the series are more closely related to future values than distant past values. This is a point of difference between TSER and TSF"
from here: [A Machine Learning Approach to
Modelling and Predicting
Biomechanical Measures of Strain
Across Bone](https://www.csse.canterbury.ac.nz/research/reports/HonsReps/2021/hons_2102.pdf)

**Comment by fkiraly** [comment]
I think in literature the names for the tasks are not consistent.
We were going to write a scientific reference where the tasks are clarified, but this is still work in progress.

Re predicting failures of assets, that would be event modelling in a continuous time series stream, no?
While predicting quality at the end of a process sounds like (early) time series regression or classification.

>So I was wondering how to choose which of this approach to use if I want to do time series regression ? I imagine that is dependent of the specific problem, but is there things to look at to make the choice ?

For choosing a method, I would look at out-of-sample performance on my given data set, adhering to best practice like validation or hold-out sets, similar to tabular regression.

As candidates, I would include:
* pipelines [feature extractor][supervised regressor] with high-performing regressors and feature extractors known to work in the domain (otherwise a base election thereof)
* pipelines [time series distance/kernel][distance/kernel based regressor]
* naive base lines
* any state-of-art methods if reportedd

There is no benchmarking study on time series regression models that I am aware of, but I also haven't conducted a systematic literature search. Maybe someone else has pointers.
